File: Examples/VisualBasic/ModifyingAndConvertingImages/ResizeImageWithResizeTypeEnumeration.vb
'
'This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Imaging for .NET API reference 
'when the project is build. Please check https:// Docs.nuget.org/consume/nuget-faq for more information. 
'If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from http://www.aspose.com/downloads, 
'Install it and then add its reference to this project. For any issues, questions or suggestions 
'Please feel free to contact us using http://www.aspose.com/community/forums/default.aspx
'


Namespace Aspose.Imaging.Examples.VisualBasic.ModifyingAndConvertingImages
    Class ResizeImageWithResizeTypeEnumeration
        Public Shared Sub Run()
            ' To get proper output please apply a valid Aspose.Imaging License. You can purchase full license or get 30 day temporary license from http:// Www.aspose.com/purchase/default.aspx.")

            ' ExStart:ResizeImageWithResizeTypeEnumeration
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_ModifyingAndConvertingImages()

            ' Load an image from disk
            Using image__1 As Image = Image.Load(dataDir & Convert.ToString("aspose-logo.jpg"))
                If Not image__1.IsCached Then
                    image__1.CacheData()
                End If

                ' specifying only width and ResizeType
                Dim newWidth As Integer = image__1.Width / 2
                image__1.ResizeWidthProportionally(newWidth, ResizeType.LanczosResample)

                ' specifying only height and ResizeType
                Dim newHeight As Integer = image__1.Height / 2
                image__1.ResizeHeightProportionally(newHeight, ResizeType.NearestNeighbourResample)
                ' saving result
                image__1.Save(dataDir & Convert.ToString("ResizeImageWithResizeTypeEnumeration_out.png"))
            End Using

            ' ExEnd:ResizeImageWithResizeTypeEnumeration
        End Sub
    End Class
End Namespace

File: Examples/CSharp/ModifyingAndConvertingImages/ResizeImageWithResizeTypeEnumeration.cs
/*
This project uses the Automatic Package Restore feature of NuGet to resolve the Aspose.Imaging for .NET API reference 
when the project is built. Please check https://learn.microsoft.com/en-us/nuget/resources/nuget-faq for more information. 
If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from https://releases.aspose.com/, 
install it, and then add its reference to this project. For any issues, questions, or suggestions, 
please feel free to contact us using https://forum.aspose.com/
*/

using System;
using Aspose.Imaging;

namespace Aspose.Imaging.Examples.CSharp.ModifyingAndConvertingImages
{
    class ResizeImageWithResizeTypeEnumeration
    {
        public static void Run()
        {
            Console.WriteLine("Running example ResizeImageWithResizeTypeEnumeration");
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_ModifyingAndConvertingImages();

            // Load an image from disk
            using (Image image = Image.Load(dataDir + "aspose-logo.jpg"))
            {
                if (!image.IsCached)
                {
                    image.CacheData();
                }

                // Specify only height, width, and ResizeType
                int newWidth = image.Width / 2;
                image.ResizeWidthProportionally(newWidth, ResizeType.LanczosResample);
                int newHeight = image.Height / 2;
                image.ResizeHeightProportionally(newHeight, ResizeType.NearestNeighbourResample);
                image.Save(dataDir + "ResizeImageWithResizeTypeEnumeration_out.png");
            }

            Console.WriteLine("Finished example ResizeImageWithResizeTypeEnumeration");
        }
    }
}